blog/2024-07-30-how-to-run-webassembly-components-on-kubernetes-with-wasmcloud/index.mdx

```diff
@@ -0,0 +1,178 @@
+* **Deploy wasmCloud applications to the cluster straight from `kubectl`**. In wasmCloud, [applications are defined in YAML manifests](../../docs/ecosystem/wadm/model) with the [OAM specification](https://oam.dev/). The manifests will look familiar to Kubernetes users&mdash;better yet, you can deploy them to the cluster the same way you would any other Kubernetes resource. 
+
+What does all of this mean in practice?
+
+1. Developers can move faster, reusing building blocks written against well-known APIs. 
+2. It's easy to run distributed apps in resource-constrained environments at the edge. 
+3. You can use wasmCloud's connectivity layer to connect Kubernetes clusters across different clouds or regions.
+4. Workloads can be scheduled dynamically across regions and clouds. 
+
+By unlocking the power of distributed components, wasmCloud extends Kubernetes as well. Let's jump in and see how it works.
+
+## Get started with wasmCloud on Kubernetes
+
+In this tutorial, we'll get wasmCloud up and running on a local Kubernetes cluster. You'll need [Helm](https://helm.sh/) and a Kubernetes cluster running on your machine. It also helps to install [wasmCloud Shell (`wash`)](/docs/installation), though it's not strictly necessary.
+
+:::info[Your friendly neighborhood Kubernetes]
+Make sure you have a Kubernetes cluster running locally. Some good options include [minikube](https://minikube.sigs.k8s.io/docs/), [kind](https://kind.sigs.k8s.io/), [k3s](https://k3s.io/), or [k0s](https://k0sproject.io/).
+:::
+
+A wasmCloud installation requires a few things to run:
+
+- **[NATS](/docs/ecosystem/nats/)** (with Jetstream enabled)
+- **[wasmCloud Application Deployment Manager (wadm)](/docs/ecosystem/wadm/)** connected to NATS
+
+If you're not sure what that means, don't worry: the following steps will help you deploy the prerequisites to your Kubernetes cluster, then deploy the operator.
+
+#### 1. Deploy prerequisites
+
+First we'll add the NATS Helm repo. If you're not familiar with it, NATS is the open source connective technology that wasmCloud uses to create a unified topology across any number of environments. 
+
+```shell
+helm repo add nats https://nats-io.github.io/k8s/helm/charts/
+```
+Next install the upstream NATS Helm chart:
+
+```shell
+helm upgrade --install -f https://raw.githubusercontent.com/wasmCloud/wasmcloud-operator/main/examples/quickstart/nats-values.yaml nats nats/nats
+```
+Deploy wadm to your Kubernetes cluster with a Helm chart hosted by wasmCloud. 
+
+```shell
+helm install wadm -f https://raw.githubusercontent.com/wasmCloud/wasmcloud-operator/main/examples/quickstart/wadm-values.yaml oci://ghcr.io/wasmcloud/charts/wadm
+```
+You can validate the installations with:
+
+```shell
+kubectl rollout status deploy,sts -l app.kubernetes.io/instance=nats
```

Comment:
Technically you should do this before you deploy `wadm`, because wadm will fail to start until NATS is deployed, but since Kubernetes takes care of restarting the pod until it does, it's also fine to do this.